src/feature/hs/hs_client.c

```diff
@@ -1445,6 +1445,79 @@ client_dir_fetch_unexpected(dir_connection_t *dir_conn, const char *reason,
                                 NULL);
 }
 
+/** Get the full filename for storing the client auth credentials for the
+ *  service in <b>onion_address</b>. The base directory is <b>dir</b>. */
+static char *
+get_client_auth_creds_filename(const char *onion_address,
+                               const char *dir)
+{
+  char *full_fname = NULL;
+  char *fname;
+
+  tor_asprintf(&fname, "%s.auth_private", onion_address);
+  full_fname = hs_path_from_filename(dir, fname);
+  tor_free(fname);
```

Comment:
Maybe you meant to use `full_name` ?

Reply:
I actually did mean `full_fname`, in the sense that it's the full path instead of just the actual filename. Let me know if you want me to change it to something else.